src/test/kotlin/com/hedvig/underwriter/graphql/GraphQLIntegrationTest.kt

```diff
@@ -0,0 +1,244 @@
+                            approveQuotes(quoteIds: ${"$"}ids)
+                        }    
+            """.trimIndent(),
+            variables = mapOf("ids" to listOf(quote.id))
+        ).assert2xx().body<Map<*, *>>()
+
+        val slot = slot<SelfChangeRequest>()
+        verify { productPricingClient.selfChangeContracts(capture(slot)) }
+        assertThat { slot.captured.quotes.first().quoteId }.isSuccess().isEqualTo(quote.id)
+        assertThat { slot.captured.memberId }.isSuccess().isEqualTo("mid")
+        assertThat { slot.captured.signSource }.isSuccess().isEqualTo(QuoteInitiatedFrom.SELF_CHANGE)
+    }
+
+    @Test
+    fun `approved quotes from CROSS_SELL get contractId and agreementId`() {
+        val quote = createQuote { initiatedFrom = "CROSS_SELL" }.body<CreateQuoteOutput>()
+
+        val contractId = UUID.randomUUID()
+        val agreementId = UUID.randomUUID()
+        every {
+            productPricingClient.createContract(any(), any())
+        } returns listOf(
+            CreateContractResponse(
+                quoteId = quote.id!!,
+                agreementId = agreementId,
+                contractId = contractId
+            )
+        )
+
+        query(
+            query = """
+                        mutation ApproveQuotes(${"$"}ids: [ID!]!) {
+                            approveQuotes(quoteIds: ${"$"}ids)
+                        }    
+            """.trimIndent(),
+            variables = mapOf("ids" to listOf(quote.id))
+        ).assert2xx().body<Map<*, *>>()
+
+        val updated = getQuote(quote.id).body<QuoteDto>()
+        assertThat { updated.contractId }.isSuccess().isEqualTo(contractId)
+        assertThat { updated.agreementId }.isSuccess().isEqualTo(agreementId)
+    }
+
+    @Test
+    fun `approved quotes from SELF_CHANGE get contractId and agreementId`() {
+        val quote = createQuote { initiatedFrom = "SELF_CHANGE" }.body<CreateQuoteOutput>()
+
+        val contractId = UUID.randomUUID()
+        val agreementId = UUID.randomUUID()
+        every {
+            productPricingClient.selfChangeContracts(any())
+        } returns SelfChangeResult(
+            updatedContracts = listOf(
+                SelfChangeResult.ContractChange(
+                    quoteId = quote.id!!,
+                    agreementId = agreementId,
+                    contractId = contractId
+                )
+            ),
+            createdContracts = listOf(),
+            terminatedContractIds = listOf()
+        )
+
+        query(
+            query = """
+                        mutation ApproveQuotes(${"$"}ids: [ID!]!) {
+                            approveQuotes(quoteIds: ${"$"}ids)
+                        }    
+            """.trimIndent(),
+            variables = mapOf("ids" to listOf(quote.id))
+        ).assert2xx().body<Map<*, *>>()
+
+        val response = getQuote(quote.id)
+        val updated = response.body<QuoteDto>()
```

Comment:
Seems like this could be compacted :) 
```suggestion
        val updated = getQuote(quote.id).body<QuoteDto>()
```